src/App.tsx

```diff
@@ -1,14 +1,49 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
+import './App.scss';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { Select } from './types/Select';
+import { filterTodos } from './components/helpers/helper';
 
 export const App: React.FC = () => {
+  const [todosFromAPI, setTodosFromAPI] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [todoChosen, setTodoChosen] = useState<Todo | null>(null);
+  const [query, setQuery] = useState('');
+  const [status, setStatus] = useState<string>('');
+
+  useEffect(() => {
+    getTodos()
+      .then((todos) => {
+        setTodosFromAPI(todos);
+      })
+      .finally(() => setIsLoading(true));
+  }, []);
+
+  const todosToRender: Todo[] = useMemo(
+    () => filterTodos(todosFromAPI, query, status as Select), [todosFromAPI, query, status],
+  );
+
+  const handleQuery = (newQuery: string) => {
+    setQuery(newQuery);
+  };
+
+  const handleSelect = (givenStatus: string) => {
+    setStatus(givenStatus);
+  };
+
+  const todoReset = () => {
```

Comment:
Function names usually start with a verb
```suggestion
  const resetTodo = () => {
```